commonmark/src/main/java/org/commonmark/internal/InlineParserImpl.java

```diff
@@ -243,17 +243,21 @@ private Node parseInline(Node previous) {
     /**
      * If RE matches at current index in the input, advance index and return the match; otherwise return null.
      */
-    private String match(Pattern re) {
+    protected String match(Pattern re) {
         if (index >= input.length()) {
             return null;
         }
-        Matcher matcher = re.matcher(input);
-        matcher.region(index, input.length());
-        boolean m = matcher.find();
-        if (m) {
-            index = matcher.end();
-            return matcher.group();
-        } else {
+        try {
+            Matcher matcher = re.matcher(input);
+            matcher.region(index, input.length());
+            boolean m = matcher.find();
+            if (m) {
+                index = matcher.end();
+                return matcher.group();
+            } else {
+                return null;
+            }
+        } catch (Error e) {
```

Comment:
Task: Seems like a good idea in general, but can we catch the specific error type here instead?